### ISSUE TITLE
A particularly difficult cycle route

### ISSUE THREAD

**Issue by RedAuburn** [organicmaps/organicmaps#9698]
This is a difficult one due to multiple small sections where bikes should dismount (but in reality usually don't), despite it being part of a cycle route.
Maybe if a path is part of a cycle relation, it should always have a small ranking increase? Related to https://github.com/organicmaps/organicmaps/issues/8028
There's also a small section tagged with `bicycle=yes`+`bicycle:advisory=dismount`, but this tag only has 40 uses. Nonetheless it would be a good fix for these cases.

<img width="300px" src="https://github.com/user-attachments/assets/3f3d5893-98ac-44d0-99e9-c91b0a85bcc8"/>

OM's route (still the same with #9692):
![image](https://github.com/user-attachments/assets/5f43fcd2-8bcd-4e4e-9be7-e32c64549968)

Valhalla and BRouter get it right, but OSRM and Graphhopper don't:
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=51.483927%2C-0.020613%3B51.483681%2C-0.007355#map=18/51.483560/-0.018719
![image](https://github.com/user-attachments/assets/f41e6c80-a8e4-492c-b67a-be26711455b3)

google gets it mostly right here (but seems to go around a cyclepath near The Sail Loft):
![image](https://github.com/user-attachments/assets/8ba9094e-9186-4a7f-96cd-bd3f6e2e0c6f)

**Comment by matkoniecz** [comment]
Wait, why cyclist should be directed there when route finding from A to B?

I would prefer to cycle on road rather than push bicycle on scenic one, unless I am specifically following scenic route rather than moving from A to B

**Comment by matkoniecz** [comment]
> There's also a small section tagged with bicycle=yes+bicycle:advisory=dismount,

Can you cycle there? Or is there sign like you liked banning cycling there?

**Comment by pastk** [comment]
I agree it looks like a matter of preference here (the alternative routes feature).

There is no need to adjust our routing to prefer the scenic route by default here.

Still the aforementioned adjustments would be good to do, just don't expect they would make OM follow exact Valhalla's option. Its more likely after adjustments it'll be like GraphHopper's.

**Comment by matkoniecz** [comment]
I am not convinced at all that `bicycle=no`  or `bicycle=dismount` (equivalents) should get boosted just because there is a bicycle route there.

Poland has hordes of badly designed bicycle routes, sometimes going through places where cycling is illegal despite preferable route being parallel to it. There is no benefit from boosting those.

**Comment by pastk** [comment]
> I am not convinced at all that `bicycle=no` or `bicycle=dismount` (equivalents) should get boosted just because there is a bicycle route there.

We're talking about a very minor increase here e.g. 10% maybe. The current "dismount" weight is 2, "boosting" it to 2.2 won't make it much different against e.g. ordinary track (weight 8).
